pkg/identity/certificate_authority.go

```diff
@@ -178,16 +179,20 @@ func (pc PeerCAConfig) Load() (*PeerCertificateAuthority, error) {
 func NewCA(ctx context.Context, opts NewCAOptions) (
 	rv *FullCertificateAuthority, err error) {
 	defer mon.Task()(&ctx)(&err)
+	var (
+		highscore uint32
+	)
 
 	if opts.Concurrency < 1 {
 		opts.Concurrency = 1
 	}
 	ctx, cancel := context.WithCancel(ctx)
 
+	log.Printf("Generating a certificate matching a difficulty of %d\n", opts.Difficulty)
 	eC := make(chan error)
 	caC := make(chan FullCertificateAuthority, 1)
 	for i := 0; i < int(opts.Concurrency); i++ {
-		go newCAWorker(ctx, opts.Difficulty, opts.ParentCert, opts.ParentKey, caC, eC)
+		go newCAWorker(ctx, i, &highscore, opts.Difficulty, opts.ParentCert, opts.ParentKey, caC, eC)
```

Comment:
Why is this highscore necessary?

Reply:
It's to give the illusion of progress. It's pass in like this so each worker can know what the current "highscore" is and display the new one to the user.